### PR TITLE
[backports-release-1.10] allow extensions to trigger from packages in [deps] (#54009)

### PR DESCRIPTION
**Pull request by topolarity**
This is a backport of #54009

### CODE REVIEW

File: base/precompilation.jl [hunk 1]
```diff
@@ -0,0 +1,1041 @@
+module Precompilation
```
Comment (KristofferC) [inline base/precompilation.jl:1]: This file should not have been added?

**Review by KristofferC — CHANGES_REQUESTED** [review]

File: test/project/Extensions/EnvWithHasExtensions/Manifest.toml [hunk 2]
```diff
@@ -18,10 +18,12 @@ version = "0.1.0"
     [deps.HasExtensions.extensions]
     Extension = "ExtDep"
     ExtensionFolder = ["ExtDep", "ExtDep2"]
+    LinearAlgebraExt = "LinearAlgebra"
```
Comment (KristofferC) [inline test/project/Extensions/EnvWithHasExtensions/Manifest.toml:21]: Should be removed?

File: test/project/Extensions/EnvWithHasExtensions/Manifest.toml [hunk 3]
```diff
@@ -18,10 +18,12 @@ version = "0.1.0"
...
 
     [deps.HasExtensions.weakdeps]
     ExtDep = "fa069be4-f60b-4d4c-8b95-f8008775090c"
     ExtDep2 = "55982ee5-2ad5-4c40-8cfe-5e9e1b01500d"
+    LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
```
Comment (KristofferC) [inline test/project/Extensions/EnvWithHasExtensions/Manifest.toml:26]: Also remove?

**Review by KristofferC — APPROVED** [review]

**Comment by topolarity** [comment]
Tests are failing due to https://github.com/JuliaLang/julia/pull/56264#issuecomment-2447449278

**Pull request merged**